### PR TITLE
Fixes to cmake build system to make installation by spack and target exporting more standard

### PR DESCRIPTION
**Pull request by womeld**
- Adds C++ headers to include directory in installation
- Moves Fortran modules to include directory in installation
- Adds a module directory to example executables to suppress build warning messages
- Revs minor version in preparation for release

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.17)
 set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
-
-project(FLCL VERSION 0.3.0 LANGUAGES Fortran C CXX)
+set(FLCL_VERSION 0.4.0)
```
Comment (dalg24) [inline CMakeLists.txt:3]: You don't need to set a variable, you can use `PROJECT_VERSION`
https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -61,60 +42,81 @@ set_target_properties(
...
+include(GNUInstallDirs)
+configure_package_config_file(
+    "cmake/flclConfig.cmake.in"
+    "${PROJECT_BINARY_DIR}/flclConfig.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake"
+)
 
 write_basic_package_version_file(
     "${PROJECT_BINARY_DIR}/flclConfigVersion.cmake"
+    VERSION "${FLCL_VERSION}"
```
Comment (dalg24) [inline CMakeLists.txt:91]: I meant you can use `${PROJECT_VERSION}` here.  I am not asking you to change though

Comment (womeld) [reply CMakeLists.txt:91]: I'm all for making things more succinct and standard.

**Comment by womeld** [comment]
Tested these fixes out, by importing the target with a CMake based application out, and it seems to finally be functioning correctly.

**Pull request merged**